internal/iso/iso.go

```
@@ -122,60 +121,84 @@ func (h *Handler) RoundTrip(req *http.Request) (*http.Response, error) {
+		return resp, nil
+	}
+	// by setting this header we are telling the logging middleware to not log its default log message.
+	// we do this because there are a lot of partial content requests.
+	resp.Header.Set("X-Global-Logging", "false")
 
 	if req.Method == http.MethodHead {
-		// Fuse client typically make a HEAD request before they start requesting content.
-		h.Logger.V(1).Info("HTTP HEAD request received, patching only occurs on 206 requests")
+		// Fuse clients typically make a HEAD request before they start requesting content.
+		log.Info("HTTP HEAD method received")
 		return resp, nil
 	}
 
-	// roundtripper should only return error when no response from the server
-	// for any other case just log the error and return 404 response
-	if resp.StatusCode == http.StatusPartialContent {
-		b, err := io.ReadAll(resp.Body)
-		if err != nil {
-			h.Logger.Error(err, "reading response bytes", "response", resp.Body)
-			return &http.Response{
-				Status:     fmt.Sprintf("%d %s", http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError)),
-				StatusCode: http.StatusInternalServerError,
-				Body:       http.NoBody,
-				Request:    req,
-			}, nil
-		}
-		if err := resp.Body.Close(); err != nil {
-			h.Logger.Error(err, "closing response body", "response", resp.Body)
-			return &http.Response{
-				Status:     fmt.Sprintf("%d %s", http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError)),
-				StatusCode: http.StatusInternalServerError,
-				Body:       http.NoBody,
-				Request:    req,
-			}, nil
-		}
+	// 0.002% of the time we log a 206 request message.
+	// In testing, it was observed that about 3000 HTTP 206 requests are made per ISO mount.
+	// 0.002% gives us about 5, +/- 3, log messages per ISO mount.
+	// We're optimizing for showing "enough" log messages so that progress can be observed.
+	if p, _ := randomPercentage(10000); p < 0.002 {
```

An additional code comment on why swallowing this error is safe and insignificant could be useful.